### PR TITLE
Fix #1431: j.l.Class.isInstance for raw JS classes.

### LINKED ISSUE #1431: j.l.Class.isInstance does not work for raw JS classes
The following does not work as expected:
```scala
val cls = classOf[dom.HTMLElement]
println(cls.isInstance(dom.document.getElementById("foo")))
```
It always displays `false` because it tries to perform the type test as if `HTMLElement` were a Scala.js class.
We could fix this by providing a custom `isInstance` function to the type data of raw JS classes.

For raw JS traits, there's nothing we can do, since `isInstanceOf` itself is not supported.

Fixing this is a prerequisite for #1430.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
CI only. Needs to be rebased on master once #1433 is merged.

File: test-suite/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala [hunk 1]
```diff
@@ -76,4 +114,10 @@ object ReflectionTest extends JasmineTest {
 
   class RenamedTestClass
 
+  @JSName("ReflectionTestRawJSClass")
+  class ReflectionTestRawJSClass extends js.Object
+
+  trait ReflectionTestRawJSTrait extends js.Object
+  object ReflectionTestRawJSObject extends js.Object
```
Comment (gzm0) [inline test-suite/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala]: You probably wanted to use this in a test

File: test-suite/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala [hunk 2]
```diff
@@ -76,4 +114,10 @@ object ReflectionTest extends JasmineTest {
 
   class RenamedTestClass
 
+  @JSName("ReflectionTestRawJSClass")
```
Comment (gzm0) [inline test-suite/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala]: Maybe this name should actually be different.

**Comment by gzm0** [comment]
That's all

File: test-suite/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala (hunk 2 above)
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala]: It *is* different, the original name is `ReflectionTest$RefectionTestRawJSClass`.

File: test-suite/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala (hunk 1 above)
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala]: Ah yes, but then I realized that there's no way to get a hand on the `j.l.Class` representing that object, so I couldn't write the test. I forgot to remove the definition of the object.

**Comment by sjrd** [comment]
Rebased and updated (removed the useless `object` in tests).

**Pull request merged**